controller/trackerquery.go

```diff
@@ -120,38 +138,90 @@ func (c *TrackerqueryController) Update(ctx *app.UpdateTrackerqueryContext) erro
 
 // Delete runs the delete action.
 func (c *TrackerqueryController) Delete(ctx *app.DeleteTrackerqueryContext) error {
-	err := application.Transactional(c.db, func(appl application.Application) error {
-		err := appl.TrackerQueries().Delete(ctx.Context, ctx.ID)
+	_, err := login.ContextIdentity(ctx)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrUnauthorized(err.Error()))
+	}
+	err = application.Transactional(c.db, func(appl application.Application) error {
+		tq, err := appl.TrackerQueries().Load(ctx.Context, ctx.ID)
 		if err != nil {
-			cause := errs.Cause(err)
-			switch cause.(type) {
-			case remoteworkitem.NotFoundError:
-				return goa.ErrNotFound(err.Error())
-			default:
-				return goa.ErrInternal(err.Error())
-			}
+			return nil
 		}
-		return ctx.OK([]byte{})
+		return appl.TrackerQueries().Delete(ctx.Context, tq.ID)
 	})
 	if err != nil {
 		return jsonapi.JSONErrorResponse(ctx, err)
 	}
 	accessTokens := getAccessTokensForTrackerQuery(c.configuration) //configuration.GetGithubAuthToken()
 	c.scheduler.ScheduleAllQueries(ctx, accessTokens)
-	return nil
+	return ctx.NoContent()
 }
 
 // List runs the list action.
 func (c *TrackerqueryController) List(ctx *app.ListTrackerqueryContext) error {
-	err := application.Transactional(c.db, func(appl application.Application) error {
-		result, err := appl.TrackerQueries().List(ctx.Context)
+	return application.Transactional(c.db, func(appl application.Application) error {
+		trackerqueries, err := appl.TrackerQueries().List(ctx)
 		if err != nil {
-			return goa.ErrInternal(fmt.Sprintf("Error listing tracker queries: %s", err.Error()))
+			return jsonapi.JSONErrorResponse(ctx, err)
 		}
-		return ctx.OK(result)
+		res := &app.TrackerQueryList{}
+		res.Data = ConvertTrackerQueries(appl, ctx.Request, trackerqueries)
+		return ctx.OK(res)
 	})
-	if err != nil {
-		return jsonapi.JSONErrorResponse(ctx, err)
+}
+
+// ConvertTrackerQueries from internal to external REST representation
+func ConvertTrackerQueries(appl application.Application, request *http.Request, trackerqueries []remoteworkitem.TrackerQuery) []*app.TrackerQuery {
```

Comment:
@DhritiShikhar How about `ConvertToAppTrackerQueries` as the name of this function?

Reply:
`ConvertToAppTrackerQueries` sounds a bit odd but it makes sense if we also have

`ConvertToAppSpace`
`ConvertToAppWorkItem`
`ConvertToAppArea`

you get the idea.

Reply:
Let's keep it `ConvertTrackerQueries` because currently we have:
`ConvertWorkItems`, `ConvertWorkItem`

Reply:
> Let's keep it `ConvertTrackerQueries` because currently we have:
> `ConvertWorkItems`, `ConvertWorkItem`

That is probably the worst reasons to come up with. The purpose is not to keep consistency at all costs with internal names but to improve over our mistakes from the past. I always have to look up what `ConvertWorkItem` does. If only it was called `ConvertWorkItemToApp`. Probably `App` is also a wrong word here.

Reply:
okay. Let's go with `ConvertTrackerQueriesToApp` for now.

Reply:
done here 
e72c82f